http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -53,14 +78,81 @@ class GKEInterpreter[F[_]: Parallel: ContextShift](
       getApp <- F.fromOption(getAppOpt, AppNotFoundException(msg.project, msg.appName, ctx.traceId))
       gkeClusterId = getApp.cluster.getGkeClusterId
       namespaceName = getApp.app.appResources.namespace.name
+      googleProject = getApp.cluster.googleProject
+      gsa = getApp.app.googleServiceAccount
+      ksaName = KubernetesServiceAccountName(s"${namespaceName.value}-ksa")
 
       _ <- kubeService.createNamespace(gkeClusterId, KubernetesNamespace(namespaceName))
       secrets <- getSecrets(namespaceName)
       _ <- secrets.parTraverse(secret =>
         kubeService.createSecret(gkeClusterId, KubernetesNamespace(namespaceName), secret)
       )
+
+      // Create KSA
+      annotations = Map("iam.gke.io/gcp-service-account" -> gsa.value)
+      _ <- kubeService
+        .createServiceAccount(
+          gkeClusterId,
+          KubernetesServiceAccount(ServiceAccountName(ksaName.value), annotations),
+          KubernetesNamespace(namespaceName)
+        )
+        .handleErrorWith {
```

Comment:
Probably don't want `handleErrorWith` since that'll recover from the error. Maybe best to just let the error fall through -- it'll be caught and logged by `LeoPubsubMessageSubscriber`

Reply:
sounds good, will update!